test/lib/server/services/run/RunService.test.js

```diff
@@ -310,4 +310,124 @@ module.exports = () => {
             { runPatch: { runQuality: RunQualities.TEST } },
         );
     });
+
+    it('should successfully update run with eorReasons with reasonTypeId', async () => {
+        const runNumber = 1;
+
+        const eorReasons = [
+            {
+                reasonTypeId: 1,
+                description: 'test1',
+            },
+            {
+                reasonTypeId: 1,
+                description: 'test2',
+            },
+        ];
+
+        const run = await runService.update(
+            { runNumber },
+            { relations: { eorReasons: eorReasons } },
+        );
+
+        expect(run.eorReasons).to.lengthOf(2);
```

Comment:
Please also check the actual EoR reasons here, their ID & title/category for all tests

Reply:
I added this in the tests I made by checking the description, since I cannot check if the whole object is included in the run.eorReasons afterwards because it adds an id to the eorReason object in the tests and that makes that it is not equal anymore. That is why I compared the description because I made them unique in the tests.